### ISSUE TITLE
[Announcement] Qt6 Support by removing deprecated functions

### ISSUE THREAD

**Issue by basyskom-dege** [OpenAPITools/openapi-generator#8232]
Since Qt6 was released recently, it would be a good to support that. The existing Qt5 generator uses some deprecated functions that are not supported in Qt6. I replaced them with the equivalent Qt6 functions. The new functions work with Qt5 aswell, except of the `Qt::SectionSkipEmpty` Flag ( equal to 0x01) that was in the `QString` namespace in Qt5. I inserted a preprocessor variable to support both, Qt5 and Qt6:
```
#if QT_VERSION >= 0x060000
    #define SKIP_EMPTY_PARTS Qt::SkipEmptyParts
#else
    #define SKIP_EMPTY_PARTS QString::SkipEmptyParts
#endif
```
I created a PR with those changes. Since the generator is called `cpp-qt5-generator` it would make sense to add a new Qt6 generator or rename it.

**Comment by wing328** [comment]
What about renaming the generator from `cpp-qt5-client` to `cpp-qt-client` to start with as the upcoming release is v5.0.0 major release and we can still include breaking changes (generator renaming)? That will require you to update your PR later though.

**Comment by basyskom-dege** [comment]
Yes renaming would be better than a second generator. I guess that wont be a big task and changing that PR later on wont be neither. I will take a further look on renaming on monday.

**Comment by wing328** [comment]
OK. I will try to come up with a PR over the weekend.

**Comment by basyskom-dege** [comment]
should i wait for the PR #8234 to be finished before renaming?

**Comment by MartinDelille** [comment]
We might close this issue since #8234 is fixed.
